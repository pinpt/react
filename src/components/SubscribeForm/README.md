### A component for the new user subscription form

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-subscribeform) | [Github](https://github.com/pinpt/react/tree/master/src/components/SubscribeForm)

#### Theming

The parent className for styling this component is `.Pinpoint.SubscribeForm`

<details>
	<summary>Base Theme Styles</summary>

```css
.Pinpoint.SubscribeForm.wrapper {
	@apply block;
}

.Pinpoint.SubscribeForm .inner {
	@apply flex flex-row;
}

.Pinpoint.SubscribeForm .inner .input {
	@apply relative border-2 mr-2 focus-within:border-blue-500 disabled:focus-within:border-gray-500;
}

.Pinpoint.SubscribeForm .spinner {
	@apply absolute top-2 right-1 text-indigo-500 z-10;
}

.Pinpoint.SubscribeForm .spinner.visible {
	@apply visible;
}

.Pinpoint.SubscribeForm .spinner.invisible {
	@apply invisible;
}

.Pinpoint.SubscribeForm input {
	@apply p-1 focus:outline-none pr-4 bg-white disabled:text-gray-400 disabled:bg-white disabled:cursor-not-allowed;
}

.Pinpoint.SubscribeForm button {
	@apply border rounded text-gray-700 bg-gray-100 px-2 py-1 disabled:bg-gray-50 disabled:text-gray-400 disabled:cursor-not-allowed;
}

.Pinpoint.SubscribeForm .success {
	@apply text-green-600 dark:text-green-500 text-sm text-left mt-1;
}

.Pinpoint.SubscribeForm .error {
	@apply text-red-500 text-sm text-left mt-1;
}

.Pinpoint.SubscribeForm .error .icon,
.Pinpoint.SubscribeForm .success .icon {
	@apply mr-1;
}

.Pinpoint.SubscribeForm .error.visible {
	@apply visible;
}

.Pinpoint.SubscribeForm .error.invisible {
	@apply invisible;
}
```

</details>
