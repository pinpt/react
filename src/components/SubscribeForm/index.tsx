import React, { useCallback, useEffect, useState } from 'react';
import { faCheckCircle, faExclamationCircle, faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { subscribe } from '../../lib/data/subscription';
import { isSubscriberCookieSet, validateEmail } from '../../lib/subscription';

import type { IPinpointConfig } from '../../lib/types';

export interface SubscribeFormProps {
	config: IPinpointConfig;
	className?: string;
	inputClassName?: string;
	buttonClassName?: string;
	successClassName?: string;
	errorClassName?: string;
	spinnerClassName?: string;
	buttonText?: React.ReactNode;
	renderError?: (error: string) => React.ReactNode;
	renderSuccess?: () => React.ReactNode;
	autoFocus?: boolean;
	placeholder?: string;
	onSuccess?: (email: string) => void | Promise<void>;
}

const defaultRenderSuccess = () => (
	<div className="flex flex-row items-center">
		<FontAwesomeIcon icon={faCheckCircle} fixedWidth className="icon" />
		Subscribed!
	</div>
);

const defaultRenderError = (error: string) => (
	<div className="flex flex-row items-center">
		<FontAwesomeIcon icon={faExclamationCircle} fixedWidth className="icon" />
		{error}
	</div>
);

const SubscribeForm = (props: SubscribeFormProps) => {
	const {
		config,
		className = '',
		inputClassName = '',
		buttonClassName = '',
		errorClassName = '',
		successClassName = '',
		spinnerClassName = '',
		buttonText = 'Subscribe',
		placeholder = 'Your email address',
		autoFocus = false,
		renderError = defaultRenderError,
		renderSuccess = defaultRenderSuccess,
		onSuccess,
	} = props;
	const [email, setEmail] = useState('');
	const [disabled, setDisabled] = useState(true);
	const [saving, setSaving] = useState(false);
	const [success, setSuccess] = useState(false);
	const [error, setError] = useState('');
	const [hidden, setHidden] = useState(false);
	const [ready, setReady] = useState(false);
	useEffect(() => {
		setHidden(isSubscriberCookieSet());
		setReady(true);
	}, []);
	useEffect(() => {
		setDisabled(!validateEmail(email));
		if (!email) {
			setError('');
		}
	}, [email]);
	const onSubmit = useCallback(async () => {
		setSaving(true);
		try {
			await subscribe(config, email);
			setSuccess(true);
			onSuccess?.(email);
			setEmail('');
		} catch (ex: any) {
			setError(ex.message);
			setSuccess(false);
		} finally {
			setSaving(false);
		}
	}, [email]);
	const _hidden = ready ? !success && hidden : true;
	return (
		<div className={`Pinpoint SubscribeForm wrapper ${_hidden ? 'invisible' : 'visible'} ${className}`}>
			<div className="inner">
				<div className="input">
					<input
						type="text"
						value={email}
						onChange={(e) => setEmail(e.target.value)}
						autoComplete="email"
						className={inputClassName}
						autoFocus={autoFocus}
						disabled={saving || success}
						placeholder={placeholder}
					/>
					<FontAwesomeIcon
						icon={faSpinner}
						pulse
						className={`spinner ${saving ? 'visible' : 'invisible'} ${spinnerClassName}`}
					/>
				</div>
				<button onClick={onSubmit} className={buttonClassName} disabled={disabled || saving || success}>
					{buttonText}
				</button>
			</div>
			{success ? (
				<div className={`success ${successClassName}`}>{renderSuccess()}</div>
			) : (
				<div className={`error ${error ? 'visible' : 'invisible'} ${errorClassName}`}>{renderError(error)}</div>
			)}
		</div>
	);
};

export default SubscribeForm;
