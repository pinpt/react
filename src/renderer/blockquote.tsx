import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
