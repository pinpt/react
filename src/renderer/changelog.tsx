import { ReactYouTubeLite as Youtube } from 'react-youtube-lite';
import { getFileUrl } from '../lib/file_metadata';
import { slugifyContent } from '../lib/string';
import { ChangelogCoverMedia, ChangelogMediaType } from '../lib/types';
import { Content as Doc } from './';

const ImageMedia = ({ src, title = '' }: { src: string; title?: string }) => {
	return (
		<div className="image">
			<img src={src} alt={title} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMediaSSR = ({ id, poster = 'hqdefault' }: { id: string; poster?: string }) => {
	// the component isn't SSR friendly so this is basically used to sort of mimic it for generated code
	const posterUrl = `https://i.ytimg.com/vi/${id}/${poster}.jpg`;
	return (
		<>
			{poster && <link rel="preload" href={posterUrl} as="image" />}
			<div
				className="youtube ryt-lite embed-responsive aspect-ratio-16/9"
				style={{ backgroundImage: `url(${posterUrl})` }}
			>
				<div className="lty-playbtn" data-url={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`} />
			</div>
		</>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
	staticMode,
}: {
	id: string;
	metadata?: Record<string, any>;
	staticMode?: true;
}) => {
	return (
		<div className="youtube">
			{!staticMode ? (
				<Youtube url={`https://www.youtube.com/watch?v=${id}`} poster={metadata?.poster} />
			) : (
				<YoutubeMediaSSR id={id} poster={metadata?.poster} />
			)}
		</div>
	);
};

const Media = ({ media, title, staticMode }: { media?: ChangelogCoverMedia; title?: string; staticMode?: true }) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactElement;
	switch (media.type) {
		case ChangelogMediaType.None: {
			return <></>;
		}
		case ChangelogMediaType.Image: {
			content = <ImageMedia src={getFileUrl(media.value)} title={title} />;
			break;
		}
		case ChangelogMediaType.Video: {
			content = <VideoMedia src={getFileUrl(media.value)} />;
			break;
		}
		case ChangelogMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} staticMode={staticMode} />;
			break;
		}
	}
	return <div className="media-container">{content}</div>;
};

interface ChangelogProps {
	id: string;
	title?: string;
	content?: any;
	coverMedia?: ChangelogCoverMedia;
	limit?: number;
	divider?: boolean;
	staticMode?: true;
}

const ChangelogLog = (props: ChangelogProps) => {
	return (
		<article className="changelog notebook-editor read-only">
			{props.coverMedia && (
				<section className="covermedia">
					<Media media={props.coverMedia} title={props.title} staticMode={props.staticMode} />
				</section>
			)}
			<section className="ProseMirror">
				<Doc node={props.content} limit={props.limit} />
				{props.limit && props.content.content?.length > props.limit && (
					<div className="continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="divider" />}
		</article>
	);
};

export default ChangelogLog;
