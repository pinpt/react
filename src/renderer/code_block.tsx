import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { twilight } from 'react-syntax-highlighter/dist/cjs/styles/prism';
import { NodeProps, registerNode } from './register';

const CodeBlock = ({ node }: NodeProps) => {
	return (
		<SyntaxHighlighter
			language={node.attrs.language}
			style={twilight}
			customStyle={{
				background: 'black',
				fontFamily: 'ui-monospace',
				borderWidth: '0px',
				borderRadius: '.25em',
			}}
		>
			{node.content?.[0].text ?? ''}
		</SyntaxHighlighter>
	);
};

registerNode('code_block', (node) => <CodeBlock node={node} />);
