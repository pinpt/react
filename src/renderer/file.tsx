import {
	faDownload, faFile, faFileAlt, faFileArchive, faFileAudio, faFileCode, faFileExcel, faFilePdf,
	faFilePowerpoint, faFileVideo, faFileWord
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NodeProps, registerNode } from './register';

export const iconForType = (type = '') => {
	if (/pdf/.test(type)) {
		return faFilePdf;
	}
	if (/zip|tar/.test(type)) {
		return faFileArchive;
	}
	if (/video/.test(type)) {
		return faFileVideo;
	}
	if (/audio/.test(type)) {
		return faFileAudio;
	}
	if (/excel|spreadsheet/.test(type)) {
		return faFileExcel;
	}
	if (/powerpoint/.test(type)) {
		return faFilePowerpoint;
	}
	if (/word/.test(type)) {
		return faFileWord;
	}
	if (/css|csv|graphql|html|java|json|php|python|typescript|xml/.test(type)) {
		return faFileCode;
	}
	if (/text/.test(type)) {
		return faFileAlt;
	}
	return faFile;
};

export const formattedSize = (size = 0) => {
	const kB = size / 1000;
	if (kB < 10) {
		return `${Math.round(kB * 10) / 10} KB`;
	}
	if (kB < 1000) {
		return `${Math.round(kB)} KB`;
	}
	const MB = kB / 1000;
	return `${Math.round(MB * 10) / 10} MB`;
};

const File = ({ node }: NodeProps) => {
	const { description, filename, publicUrl, size, type } = node.attrs;

	const onDownload = () => {
		if (publicUrl) {
			const u = new URL(publicUrl);
			u.searchParams.set('filename', filename);
			u.searchParams.set('disposition', 'attachment');
			const a = document.createElement('a');
			a.style.display = 'none';
			a.href = u.toString();
			// the filename you want
			a.download = filename;
			document.body.appendChild(a);
			a.click();
			a.remove();
		}
	};

	return (
		<div className="file">
			<div className="container">
				<FontAwesomeIcon icon={iconForType(type)} />
				<div className="details">
					{publicUrl ? (
						<a href={publicUrl} target="_blank" rel="noopener noreferrer" className="filename">
							{filename}
						</a>
					) : (
						<div className="filename">{filename}</div>
					)}
					{description && <div className="description">{description}</div>}
					<div className="size">{formattedSize(size)}</div>
				</div>
				<div className="options">
					{publicUrl && (
						<div className="option group" onClick={onDownload}>
							<FontAwesomeIcon icon={faDownload} />
							<span>Download</span>
						</div>
					)}
				</div>
			</div>
		</div>
	);
};

registerNode('file', (node) => <File node={node} />);
