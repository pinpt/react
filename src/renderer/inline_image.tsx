import { extractImageMetadataFromFileID } from '../lib/file_metadata';
import { NodeProps, registerNode } from './register';

const InlineImage = ({ node }: NodeProps) => {
	const { size } = extractImageMetadataFromFileID(node.attrs?.src);
	const width = size?.width;
	const height = size?.height;

	const image = (
		<img src={node.attrs?.src} width={width} height={height} alt={node.attrs?.alt} className="inline_image" />
	);

	return (
		<div>
			{node.attrs?.href ? (
				<a href={node.attrs.href} target="_blank" rel="noopener noreferrer">
					{image}
				</a>
			) : (
				<>{image}</>
			)}
		</div>
	);
};

registerNode('inline-image', (node) => <InlineImage node={node} />);
