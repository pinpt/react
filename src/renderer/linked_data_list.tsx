import { NodeProps, recurseIntoChildren, registerNode } from './register';

const LinkedDataList = ({ node }: NodeProps) => {
	return <div className="linked-data-list">{recurseIntoChildren(node)}</div>;
};

registerNode('linked_data_list', (node) => <LinkedDataList node={node} />);
