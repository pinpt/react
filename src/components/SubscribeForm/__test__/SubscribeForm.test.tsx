import renderer from 'react-test-renderer';
import SubscribeForm from '../';
import type { IPinpointConfig } from '../../../lib/types';

const config = {
	slug: 'pinpoint',
	siteId: 'PirxVTE94u3YmGNOySRY',
	apihost: 'api.pinpoint.com',
	pageSize: 11,
	siteUrl: 'https://changelog.pinpoint.com',
	apiKey: '1234',
} as IPinpointConfig;

test('Test default state', () => {
	const component = renderer.create(<SubscribeForm config={config} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom placeholder', () => {
	const component = renderer.create(<SubscribeForm config={config} placeholder="Email Address" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom button text', () => {
	const component = renderer.create(<SubscribeForm config={config} buttonText="Do it" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
