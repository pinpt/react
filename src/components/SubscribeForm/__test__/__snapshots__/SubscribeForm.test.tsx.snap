// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom button text 1`] = `
<div
  className="Pinpoint SubscribeForm wrapper invisible "
>
  <div
    className="inner"
  >
    <div
      className="input"
    >
      <input
        autoComplete="email"
        autoFocus={false}
        className=""
        disabled={false}
        onChange={[Function]}
        placeholder="Your email address"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spinner fa-w-16 fa-pulse spinner invisible "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <button
      className=""
      disabled={true}
      onClick={[Function]}
    >
      Do it
    </button>
  </div>
  <div
    className="error invisible "
  >
    <div
      className="flex flex-row items-center"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 fa-fw icon"
        data-icon="exclamation-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      
    </div>
  </div>
</div>
`;

exports[`Test custom placeholder 1`] = `
<div
  className="Pinpoint SubscribeForm wrapper visible "
>
  <div
    className="inner"
  >
    <div
      className="input"
    >
      <input
        autoComplete="email"
        autoFocus={false}
        className=""
        disabled={false}
        onChange={[Function]}
        placeholder="Email Address"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spinner fa-w-16 fa-pulse spinner invisible "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <button
      className=""
      disabled={true}
      onClick={[Function]}
    >
      Subscribe
    </button>
  </div>
  <div
    className="error invisible "
  >
    <div
      className="flex flex-row items-center"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 fa-fw icon"
        data-icon="exclamation-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      
    </div>
  </div>
</div>
`;

exports[`Test default state 1`] = `
<div
  className="Pinpoint SubscribeForm wrapper invisible "
>
  <div
    className="inner"
  >
    <div
      className="input"
    >
      <input
        autoComplete="email"
        autoFocus={false}
        className=""
        disabled={false}
        onChange={[Function]}
        placeholder="Your email address"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spinner fa-w-16 fa-pulse spinner invisible "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <button
      className=""
      disabled={true}
      onClick={[Function]}
    >
      Subscribe
    </button>
  </div>
  <div
    className="error invisible "
  >
    <div
      className="flex flex-row items-center"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 fa-fw icon"
        data-icon="exclamation-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      
    </div>
  </div>
</div>
`;
